src/Cake.Common/Tools/MSBuild/MSBuildResolver.cs

```diff
@@ -205,16 +222,21 @@ private static DirectoryPath GetVisualStudioPath(ICakeEnvironment environment, M
         }
 
         private static DirectoryPath GetVisualStudio2022Path(IFileSystem fileSystem, ICakeEnvironment environment,
-            MSBuildPlatform buildPlatform)
+            MSBuildPlatform buildPlatform, bool allowPreviewVersion)
         {
-            var vsEditions = new[]
+            var vsEditions = new List<string>
             {
                 "Enterprise",
                 "Professional",
                 "Community",
                 "BuildTools",
             };
 
+            if (allowPreviewVersion)
+            {
+                vsEditions.Insert(0, "Preview");
```

Comment:
Shouldn't `Preview` be the last option / fallback if none of the other editions were found? 🤔 

```suggestion
                vsEditions.Add("Preview");
```

Reply:
I don't think so. 2022 preview currently is 17.1.x, the others are 17.0.x. 

If everything is up-to-date preview is the highest available version.